source/extensions/load_balancing_policies/maglev/config.cc

```diff
@@ -15,24 +15,26 @@ Factory::create(OptRef<const Upstream::LoadBalancerConfig> lb_config,
                 const Upstream::PrioritySet& priority_set, Runtime::Loader& runtime,
                 Random::RandomGenerator& random, TimeSource&) {
 
-  const auto* typed_lb_config =
-      dynamic_cast<const Upstream::LoadBalancerConfigWrapper*>(lb_config.ptr());
-  auto typed_proto_config = typed_lb_config == nullptr
-                                ? OptRef<const MaglevLbProto>{}
-                                : typed_lb_config->typedProtoConfig<MaglevLbProto>();
+  const auto* typed_lb_config = dynamic_cast<const Upstream::TypedMaglevLbConfig*>(lb_config.ptr());
+
+  const auto* legacy_lb_config =
+      dynamic_cast<const Upstream::LegacyTypedMaglevLbConfig*>(lb_config.ptr());
```

Comment:
Is it better to move those `dynamic_cast` to a method in the lb_config?

Reply:
I don't think so. Note these dynamic cast is only necessary for the built-in LBs, like the round robin, random.
But the `LoadBalancerConfig` interface is not only for the built-in LBs.